internal/colorize/cropped.go

```diff
@@ -1,61 +1,44 @@
 package colorize
 
-type Entry struct {
+type CroppedLines []CroppedLine
+
+type CroppedLine struct {
 	Line   string
 	Length int
 }
 
-func GetCroppedText(text string, maxLen int) []Entry {
-	var line string // the line we're building, including the color codes
-	var plainTextPosition int
-	var plainTextWritten int
-
-	entries := make([]Entry, 0)
-	colorCodes := colorRx.FindAllStringSubmatchIndex(text, -1)
-	runeText := []rune(text)
-
-	for plainTextPosition < len(runeText) {
-		// If we reach an index that we recognize (ie. the start of a tag)
-		// then we write the whole tag, otherwise write by rune
-		for _, match := range colorCodes {
-			start, stop := match[0], match[1]
-			if plainTextPosition == start {
-				line += string(runeText[plainTextPosition:stop])
-				plainTextPosition = stop
-			}
-		}
-
-		if plainTextPosition > len(runeText)-1 {
-			entries = append(entries, Entry{line, plainTextWritten})
-			break
-		}
+func GetCroppedText(text []rune, maxLen int) CroppedLines {
+	entries := make([]CroppedLine, 0)
+	colorCodes := colorRx.FindAllStringSubmatchIndex(string(text), -1)
 
-		// Reached end of line
-		if plainTextWritten == maxLen {
-			entries = append(entries, Entry{line, plainTextWritten})
-			plainTextWritten = 0
-			line = ""
-			continue
-		}
+	entry := CroppedLine{}
+	for pos := 0; pos < len(text); pos++ {
+		inColorTag := inRange(pos, colorCodes)
+		amend := text[pos]
+		lineEnd := amend == '\n'
 
-		// Text already has line ending, so terminate the line here
-		if runeText[plainTextPosition] == '\n' {
-			plainTextPosition++
-			entries = append(entries, Entry{line, plainTextWritten})
-			plainTextWritten = 0
-			line = ""
-			continue
+		if !lineEnd {
+			entry.Line += string(amend)
+			if !inColorTag {
+				entry.Length++
+			}
 		}
 
-		if plainTextPosition == len(runeText)-1 {
-			line += string(runeText[plainTextPosition])
-			entries = append(entries, Entry{line, plainTextWritten + 1})
+		if !inRange(pos+1, colorCodes) && (entry.Length == maxLen || lineEnd || pos == len(text)-1) {
```

Comment:
Nit: This or a similar comment may be helpful
```suggestion
                // Ensure the next position is not within a color tag and check conditions that would end this entry
		if !inRange(pos+1, colorCodes) && (entry.Length == maxLen || lineEnd || pos == len(text)-1) {
```